bdsg/include/bdsg/snarl_distance_index.hpp

```diff
@@ -787,10 +785,185 @@ class SnarlDistanceIndex : public SnarlDecomposition, public TriviallySerializab
+    const static size_t TRIVIAL_SNARL_COMPONENT_OFFSET = 7;
+   
+    /*Snarl record (which occurs within a chain)
+     * 
+     * - A snarl record for each snarl, which are stuck in chains
+     *   [snarl tag, # nodes, pointer to parent, min length, max length, rank in parent, 
+     *      pointer to children (in child vector), distance vector]
+     *   The rank of the start node will always be 0 and the end node rank will be 1
+     *   start/end are the start and end nodes, include the orientations
+     *   For the first and last nodes, we only care about the node sides pointing in
+     *   Each node side in the actual distance matrix will be 2*(rank-1) for the left side, and 
+     *   2rank+1 for the right side, and 0 for the start, 1 for the end, where we only keep the 
+     *   inner node side of the start and end
+     *   Node count is the number of nodes, not including boundary nodes
+     */
+    const static size_t SNARL_RECORD_SIZE = 8;
+    const static size_t SNARL_NODE_COUNT_OFFSET = 1;
+    const static size_t SNARL_PARENT_OFFSET = 2;
+    const static size_t SNARL_MIN_LENGTH_OFFSET = 3;
+    const static size_t SNARL_MAX_LENGTH_OFFSET = 4;
+    const static size_t SNARL_DISTANCE_START_START_OFFSET = 5;
+    const static size_t SNARL_DISTANCE_END_END_OFFSET = 6;
+    const static size_t SNARL_CHILD_RECORD_OFFSET = 7;
+
+    /*A simple snarl for bubbles with only nodes with two edges, one to each bound
+     * [simple snarl tag, node count+length, parent, [node id, node length]xN
+    */
+    const static size_t SIMPLE_SNARL_RECORD_SIZE = 3;
+    //This one stores the node count and min and max lengths of the snarl
+    //It'll take 26 bits: 11 for each length
+    const static size_t SIMPLE_SNARL_NODE_COUNT_AND_LENGTHS_OFFSET = 1;
+    const static size_t SIMPLE_SNARL_PARENT_OFFSET = 2;
+
+     /*  At the end is the (single) child vector, listing children in snarls
+     *   [child vector tag, (pointer to records) x N
+     *   Each snarl will have a pointer into here, and will also know how many children it has
+     */ 
+
+private:
+    /*Give each of the enum types a name for printing */
+    vector<std::string> record_t_as_string = {"ROOT", "NODE", "DISTANCED_NODE", 
+                     "TRIVIAL_SNARL", "DISTANCED_TRIVIAL_SNARL",
+                     "SNARL", "DISTANCED_SNARL", "SIMPLE_SNARL", "OVERSIZED_SNARL", 
+                     "ROOT_SNARL", "DISTANCED_ROOT_SNARL",
+                     "CHAIN", "DISTANCED_CHAIN", "MULTICOMPONENT_CHAIN",
+                     "CHILDREN"};
+    vector<std::string> connectivity_t_as_string = { "START_START", "START_END", "START_TIP", 
+                            "END_START", "END_END", "END_TIP", 
+                            "TIP_START", "TIP_END", "TIP_TIP"};
+    vector<std::string> net_handle_record_t_string = {"ROOT_HANDLE", "NODE_HANDLE", "SNARL_HANDLE", 
+                                                "CHAIN_HANDLE", "SENTINEL_HANDLE"};
```

Comment:
These should probably be `static const`. They could also just be `static const char*[]` or something, instead of full heap-allocated vectors and strings.